### DISCUSSION TITLE
Has github action somthing like travis_fold?

### DISCUSSION THREAD

**Discussion by prisis** [community/community#25618] [Actions]
<p>Hey :), </p>
<p>in travis you can use something like a travis_fold <a href="https://gist.github.com/prisis/e050c4da44c6ee7fa1519912eac19563" rel="nofollow noopener">https://gist.github.com/prisis/e050c4da44c6ee7fa1519912eac19563</a> to show steps inside the log.</p>
<p>Than there would be a way to build dynamic step text outputs.</p>
<p>Will something like this be supported in the future or is it support?</p>

**Comment by wipe2238** [comment, marked as answer]
<p>There is ::group “command” which allows to hide boring parts of scripts. Can’t remember where i found this; it’s not in <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/development-tools-for-github-actions" rel="nofollow noopener">docs</a>, so either it was used by someone else in their workflow, or inside ‘actions/toolkit’ repo.</p>
<p>Anyway.</p>
<pre><code>- run: |
       echo Important stuff
       echo ::group::Boring stuff
       echo very
       echo boring
       echo stuff
       echo ::endgroup::
       echo Exciting stuff
</code></pre>
<p>will become</p>
<pre><code>Important stuff
▶ Boring stuff
Exciting stuff
</code></pre>
<p>when checking workflow run under Actions tab, with “Boring stuff” clickable to expand group.</p>
<p>There are two minor issues with ::group i know of:</p>
<ul>
<li>
<p><a href="https://github.community/t/colorizing-warnings-errors-breaks-group/16695">Colorizing warnings/errors breaks ::group</a> </p>
</li>
<li>
<p><a href="https://github.community/t/first-use-of-echo-name-in-script-does-not-work-properly/16320/3">Commands writing to error stream might ‘escape’ grouping</a> (see last answer by thboop)</p>
</li>
</ul>
<p>Not sure if first one is still valid, but second one i still can see in recent runs; easy to avoid.</p>
<pre><code>- run: |
         echo ::group::Stuff
         ./name 2&gt;&amp;1
         echo ::endgroup::
  shell: bash
</code></pre>

**Comment by jjlin** [comment]
<p>The grouping feature is documented here:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#grouping-log-lines">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#grouping-log-lines" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/203380/181092229-64632f28-4a81-4f15-bf8b-02a52e40fcb3.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#grouping-log-lines" target="_blank" rel="noopener nofollow ugc">Workflow commands for GitHub Actions - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
